inc/classes/admin/class-options-data.php

```diff
@@ -58,9 +58,39 @@ public function get( $key, $default = '' ) {
 		$value = apply_filters( 'pre_get_rocket_option_' . $key, null, $default ); // phpcs:ignore WordPress.NamingConventions.PrefixAllGlobals.NonPrefixedHooknameFound
 
 		if ( null !== $value ) {
+			switch ( $key ) {
+				case 'cache_mobile':
+					if ( rocket_get_config( 'cache_mobile' ) !== $value ) {
+						rocket_generate_config_file( false );
+					}
+
+					break;
+				case 'do_caching_mobile_files':
+					if ( rocket_get_config( 'do_caching_mobile_files' ) !== $value ) {
+						rocket_generate_config_file( false );
+					}
+
+					break;
+			}
+
 			return $value;
 		}
 
+		switch ( $key ) {
+			case 'cache_mobile':
+				if ( isset( $this->options[ $key ] ) && rocket_get_config( 'cache_mobile' ) !== $this->options[ $key ] ) {
+					rocket_generate_config_file();
+				}
+
+				break;
+			case 'do_caching_mobile_files':
+				if ( isset( $this->options[ $key ] ) && rocket_get_config( 'do_caching_mobile_files' ) !== $this->options[ $key ] ) {
+					rocket_generate_config_file();
+				}
+
+				break;
+		}
+
```

Comment:
I don't like this, it's introducing specific coupling to a class that is mainly generic. Why is this needed and what is the goal?

Reply:
@Tabrisrp We are not regenerating the config file when the `pre_get_rocket_option` filter is used to disable mobile cache or separate cache for mobile, that was the reason for this. Do you see a better way to handle this?

Reply:
> It could be done in the callback hooked on this filter instead. Do we have use cases in the plugin?
@Tabrisrp I don't think we have use cases but since we are giving the users the freedom of disabling these options, there are chances, they won't be handling the regeneration of config file themselves except we document this during the use of these filters.

Reply:
In my opinion this should be handled in the callback implementation with the documentation helping for this specific case. Usage of hooks is for developers/advanced users, so it makes the most sense to me.